cmd/cluster/delete/runner.go

```diff
@@ -0,0 +1,54 @@
+package delete
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kind/pkg/cluster"
+)
+
+const (
+	waitForReady = 2 * time.Minute
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	var err error
+
+	{
+		kindCtx := cluster.NewContext(r.flag.Name)
+
+		err = kindCtx.Delete()
```

Comment:
Have you tried this? I would be interested what kind of errors can happen here? e.g. will it wait forever if there is a problem? Or is there a timeout?

Reply:
Yes I tried it a couple of times and I worked fine. I looked under the hood and it deletes the kubeconfig and each docker container.

Also I'm just adding the command for completeness. In e2e builds I don't think we need it. The Circle VM goes away at the end of the job and cleaning up is Circle's problem :)